### PR TITLE
React native publishing

### PR DESCRIPTION
**Pull request by emclaug2**
#### Changes proposed in this Pull Request:
- Add publishing.md & auto deploys

### CODE REVIEW

**Comment by emclaug2** [comment]
One thing to notice for this repo is that the top-level package.json controls the npm package version.  

This is different than how angular and react handles its package.json.  I think RN should be refactored a bit to remove top-level package.json version and dependencies; it should only be used for scripts.

File: PUBLISHING.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Publishing Instructions
+
+To update the version number, edit the version in `package.json`. 
+
+To publish a new package through NPM, run the following commands from the root folder: 
+
+```
+yarn build 
+cd dist
+npm publish
```
Comment (joebochill) [inline PUBLISHING.md:10]:
```suggestion
npm publish --tag <alpha | beta>
```

File: PUBLISHING.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+```
+
+> The above command should only be ran for `alpha` or `beta` packages.  This repo's CircleCI will automatically publish latest packages from the master branch.
```
Comment (joebochill) [inline PUBLISHING.md:13]:
```suggestion
> The above command should only be run for `alpha` or `beta` packages.  This repo's CircleCI will automatically publish latest packages from the master branch. 
```

**Review by joebochill — APPROVED** [review]

**Comment by joebochill** [comment]
> One thing to notice for this repo is that the top-level package.json controls the npm package version.
> 
> This is different than how angular and react handles its package.json. I think RN should be refactored a bit to remove top-level package.json version and dependencies; it should only be used for scripts.

Let's write up an enhancement for that

File: PUBLISHING.md (hunk 1 above)
Comment (emclaug2) [reply PUBLISHING.md:10]: I'm going to apply this change to all other open PRs right now.

**Review by joebochill — APPROVED** [review]

**Pull request merged**